### DISCUSSION TITLE
requireAuth and subsequent UI flow question

### DISCUSSION THREAD

**Discussion by SuperstringDev** [dexie/Dexie.js#1362] [Q&A]
Just a quick question regarding the OTP sign-in flow.

I'm placing the store setup code in it's own module then calling it in the apps index.js constructor. From the documentations example the db.cloud.configure() requireAuth parameter is shown to be part of the store setup flow. The documentation says "If this parameter is true, the system will make sure there it has a valid authentication token before any call to the database can proceed.

Looking at the React sample app the requireAuth parameter is False, so there is no authentication flow to look at there. How is my app suppose to know that this is an initial sign-in or a refresh sign-in and subsequently present the UI component that will allow the user to sign-in? How does the app then know the result sign-in status so it can abort or bring up the next UI?

P.S. I'm assuming the app just always goes to a sign-in landing page on startup and ends on the same page when a member signs-out. In such a case the app entry UI is consistent and always the same.

UPDATE

Looks like "Dexie cloud addon beta 3" has the sleek new authentication flow. Will take a good look at that when it is merged. Seeing the new changes in action in the live demo has been very helpful, thanks.

**Comment by dusty-phillips** [comment]
If you're still looking, I worked out some solutions here:

https://github.com/dfahlander/Dexie.js/issues/1386
